### PR TITLE
Adding a DaCe Debugger with the type 'dace-debug'

### PR DESCRIPTION
**Pull request by benibenj**

### CODE REVIEW

File: src/debugger/daceDebugger.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+class DaceDebugConfigProvider
+implements vscode.DebugConfigurationProvider {
+
+    resolveDebugConfiguration(
+        folder: vscode.WorkspaceFolder | undefined,
+        config: vscode.DebugConfiguration,
+        token?: vscode.CancellationToken
+    ): vscode.ProviderResult<vscode.DebugConfiguration> {
+
+        // if launch.json is missing or empty
```
Comment (tbennun) [inline src/debugger/daceDebugger.ts:33]: I think you should run an auto formatter on the file

**Review by tbennun — CHANGES_REQUESTED** [review]
Looks good, minor changes requested

File: src/debugger/daceDebugger.ts (hunk 1 above)
Comment (phschaad) [reply src/debugger/daceDebugger.ts:33]: Agreed - on this note; I am working on a rule-file for `prettier` which will be added shortly to take care of that. For now, just try to match the remainder of the project. Most notably for this file that means a line length of no more than 80 columns, and (4) space indentation (no tabs). Thanks!

File: package-lock.json [hunk 2]
```diff
@@ -1,2207 +1,4917 @@
 {
-	"name": "sdfv",
```
[the remark this answers is not in the document]
Comment (benibenj) [reply package-lock.json:2]: It seems like we aren't using the same npm version. I am running npm-v7 in which they changed the formatting of the lockfile. As I'm running the newer release, this caused the lockfile to automatically change from lockFileVersion 1 -> 2. I believe the number of dependencies is the same, however, they just added more information to "boost the performance".

I got this information from here: 
https://nitayneeman.com/posts/catching-up-with-package-lockfile-changes-in-npm-v7/#new-lockfile-format

Let me know which version you prefer to use. If you keep the older lockfile version, then we can just remove the changes to this package-lock file and use the file that's currently deployed.

[the remark this answers is not in the document]
Comment (tbennun) [reply package-lock.json:2]: @phschaad objections?

File: src/debugger/daceDebugger.ts (hunk 1 above)
Comment (benibenj) [reply src/debugger/daceDebugger.ts:30]: I ran a formatter on the file and made sure to not surpass the 80 column line in my latest commit.

File: package-lock.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (phschaad) [reply package-lock.json:2]: Upgrading to npm-v7 works for me, that seems reasonable. Thanks for the explanation!

File: src/debugger/daceDebugger.ts (hunk 1 above)
Comment (phschaad) [reply src/debugger/daceDebugger.ts:30]: LGTM, thanks!

**Review by tbennun — APPROVED** [review]
LGTM!

**Pull request merged**